appserver/admingui/common/src/main/java/org/glassfish/admingui/common/handlers/NewSSLHandlers.java

```diff
@@ -126,100 +129,85 @@ public static void convertCiphersItemsToStr(HandlerContext handlerCtx) {
-                for (int i = 0; i < selectedCiphers.length; i++) {
-                    sb.append(sep).append("+").append(selectedCiphers[i]);
-                    sep = ",";
-                }
-            }
-            return sb.toString();
-        }
 
-        private static Vector getCiphersVector(String[] allCiphers){
-            Vector ciphers = new Vector();
-            if (allCiphers != null){
-                for(int i=0; i<allCiphers.length; i++){
-                    ciphers.add(allCiphers[i]);
+    private static String[] getSelectedCiphersList(String selectedCiphers){
+        List<String> selItems = new ArrayList<>();
+        if(selectedCiphers != null){
+            String[] sel = selectedCiphers.split(","); //NOI18N
+            for (String cName : sel) {
+                if (cName.startsWith("+")) { //NOI18N
+                    selItems.add(cName.substring(1));
                 }
             }
-            return ciphers;
         }
-        
-        private static String[] getCommonCiphers(Vector ciphers){
-            Vector commonCiphers = filterCiphers(ciphers, COMMON_CIPHERS);
-            String[] ciphersList = (String[])commonCiphers.toArray(new String[commonCiphers.size()]);
-            return ciphersList;
+        return selItems.toArray(new String[0]);
+    }
+
+    private static String processSelectedCiphers(String[] selectedCiphers, String ciphers){
+        StringBuilder sb = new StringBuilder();
+        String sep = "";
+        if ( ! GuiUtil.isEmpty(ciphers)){
+            sb.append(ciphers);
+            sep = ",";
         }
-        
-        private static String[] getEccCiphers(Vector ciphers){
-            Vector eccCiphers = breakUpCiphers(new Vector(), ciphers, "_ECDH_"); //NOI18N
-            eccCiphers = breakUpCiphers(eccCiphers, ciphers, "_ECDHE_"); //NOI18N
-            String[] ciphersList = (String[])eccCiphers.toArray(new String[eccCiphers.size()]);
-            return ciphersList;
-        }    
-        
-        private static String[] getEphemeralCiphers(Vector ciphers){
-            Vector ephmCiphers = breakUpCiphers(new Vector(), ciphers, "_DHE_RSA_"); //NOI18N
-            ephmCiphers = breakUpCiphers(ephmCiphers, ciphers, "_DHE_DSS_"); //NOI18N
-            String[] ciphersList = (String[])ephmCiphers.toArray(new String[ephmCiphers.size()]);
-            return ciphersList;
+        if(selectedCiphers != null){
+            for (String selectedCipher : selectedCiphers) {
+                sb.append(sep).append("+").append(selectedCipher);
+                sep = ",";
+            }
         }
-        
-        private static String[] getOtherCiphers(Vector ciphers){
-            Vector bitCiphers = filterCiphers(ciphers, BIT_CIPHERS);
-            String[] ciphersList = (String[])bitCiphers.toArray(new String[bitCiphers.size()]);
-            return ciphersList;
+        return sb.toString();
+    }
+
+    private static List<String> getCiphers(String[] allCiphers){
```

Comment:
Couldn't this method be replaced with `Arrays.asList(allCiphers)` or does the resulting `List` have to be modifiable?